contrib/win32/win32compat/w32log.c

```diff
@@ -114,17 +118,37 @@ openlog_file()
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), __wprogdata); /* "%programData%" */
 		wcscat_s(ssh_cfg_path, _countof(ssh_cfg_path), L"\\ssh"); /* "%programData%\\ssh" */
 
+		wchar_t* tmp_identity = NULL;
+		if (strcmp(identity, "sftp-server") == 0) {
+			tmp_identity = utf8_to_utf16(identity);
+			if (!tmp_identity)
+				return;
+		}
+		else {
+			tmp_identity = malloc((wcslen(tail) - 4) * sizeof(wchar_t));
```

Comment:
don't you need to add "+1" for null character?

Reply:
Yes. tail is in this format: "\sshd.exe", so when I subtract 4 there is still one extra character space for the null character.

Reply:
for better readability, 

tmp_identity = malloc((wcslen(tail)) * sizeof(wchar_t));

It's okay to allocate few extra bytes